### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by 0xdea**
Hi, thank you for sharing this fantastic tool! I've implemented `get_cmt()`, `set_cmt()`, and `append_cmt()` functionality from the IDA SDK, because I needed it for a tool that I'm developing. 

I've followed your guidelines and everything should be in order, but since it's my first contribution to this project please let me know if I've missed something. More contributions are coming.

### CODE REVIEW

File: idalib/src/idb.rs [hunk 1]
```diff
@@ -213,6 +214,36 @@ impl IDB {
...
+        }
+        Ok(())
+    }
+
+    pub fn append_cmt(
+        &self,
+        ea: Address,
+        comm: impl AsRef<str>,
+        rptble: bool,
+    ) -> Result<(), NulError> {
```
Comment (xorpse) [inline idalib/src/idb.rs:239]: I'd change this (and the other instance of `NulError`) to `IDAError` so it's easier downstream to handle errors from this library; should be possible by doing this:

```rust
let s = CString::new(...).map_err(IDAError::ffi)?;
```

File: idalib/src/idb.rs [hunk 2]
```diff
@@ -213,6 +214,36 @@ impl IDB {
...
+        let s = CString::new(comm.as_ref())?;
+        unsafe {
+            set_cmt(ea.into(), s.as_ptr(), rptble);
+        }
+        Ok(())
+    }
+
+    pub fn append_cmt(
```
Comment (xorpse) [inline idalib/src/idb.rs:234]: I'd change this to be `append_cmt_with` and have a second variant, so we can have a default API that gives us non-repeatable comment:

```
pub fn append_cmt(&self, ea: Address, comm: impl AsRef<str>) -> Result<(), ..> {
    self.append_cmt_with(ea, comm, false)
}
```

File: idalib/src/idb.rs [hunk 3]
```diff
@@ -213,6 +214,36 @@ impl IDB {
         }
     }
 
+    pub fn get_cmt(&self, ea: Address, rptble: bool) -> String {
```
Comment (xorpse) [inline idalib/src/idb.rs:217]: Similar to `append_cmt`/`set_cmt`, I'd create a second variation of this API with a default value for non-repeatable comments.

File: idalib/src/idb.rs [hunk 4]
```diff
@@ -213,6 +214,36 @@ impl IDB {
...
+        unsafe { idalib_get_cmt(ea.into(), rptble) }
+    }
+
+    pub fn set_cmt(
```
Comment (xorpse) [inline idalib/src/idb.rs:221]: Similar to `append_cmt`, I'd create a second variation of this API with a default value for non-repeatable comments.

**Review by xorpse — COMMENTED** [review]
Thanks a lot for your contribution! I've left some comments to improve the ergonomics of the API a bit, but in general the PR is great--thanks again :).

**Comment by 0xdea** [comment]
Thank you for your review! I've implemented the recommended changes, it should be good to go now. Cheers.

**Review by xorpse — APPROVED** [review]
Looks good, thank you! I'll prepare a new release to crates.io over the weekend

**Pull request merged**